qutip/tests/core/data/test_mathematics.py

```diff
@@ -92,6 +92,18 @@ def shapes_binary_bad_matmul(dim=100):
     ]
 
 
+def shapes_square(dim=100):
+    return [
+        (pytest.param((1, 1), id="1"),),
+        (pytest.param((dim, dim), id="100"),),
+    ]
+
+
+def shapes_bad_square(dim=100):
+    return [
+        (x,) for x in shapes_unary() if x.values[0][0] != x.values[0][1]
+    ]
+
```

Comment:
Why the `dim` input? In `shapes_square`, if it is use, the `id` will not reflect it. In `shapes_bad_square`, it is not used.